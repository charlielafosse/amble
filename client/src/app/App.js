import React, { Component } from "react";
import { Route, Switch, BrowserRouter } from "react-router-dom";
import "./App.css";
import Header from "./components/header";
import Home from "./pages/home";
import AddWalk from "./pages/addWalk";
import Walks from "./pages/walks";

class App extends Component {
  render() {
    const App = () => (
      <div>
        <Header title="amble" />
        <BrowserRouter>
          <Switch>
            <Route exact path="/" component={Home} />
            <Route path="/addWalk" component={AddWalk} />
            <Route path="/walks" component={Walks} />
          </Switch>
        </BrowserRouter>
      </div>
    );
    return <App />;
  }
}

export default App;
