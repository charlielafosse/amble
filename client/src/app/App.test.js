import React from "react";
import renderer from "react-test-renderer";
import App from "./App";

test("App renders properly", () => {
  const component = renderer.create(<App />).toJSON();
  expect(component).toMatchSnapshot();
});
