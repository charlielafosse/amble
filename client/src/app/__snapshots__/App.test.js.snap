// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders properly 1`] = `
<div>
  <nav
    className="header"
  >
    <a
      href="/"
    >
      amble
    </a>
    <ul>
      <li>
        <a
          href="/"
        >
          about
        </a>
      </li>
      <li>
        <a
          href="/walks"
        >
          walks
        </a>
      </li>
      <li>
        <a
          href="/"
        >
          login
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="App"
  >
    <h1>
      Project Home
    </h1>
    <a
      href="/addWalk"
      onClick={[Function]}
    >
      <button
        variant="raised"
      >
        Add a walk
      </button>
    </a>
    <a
      href="/walks"
      onClick={[Function]}
    >
      <button
        variant="raised"
      >
        See walks
      </button>
    </a>
  </div>
</div>
`;
